pkg/retrier/retrier.go

```diff
@@ -63,9 +63,14 @@ func WithRetryPolicy(policy RetryPolicy) RetrierOpt {
 // Retry runs the fn function until it either successful completes (not error),
 // the set timeout reached or the retry policy aborts the execution
 func (r *Retrier) Retry(fn func() error) error {
+	if r == nil {
+		return fn()
+	}
+
 	start := time.Now()
 	retries := 0
 	var err error
+	logger.V(5).Info("Retrier:", "timeout", r.timeout, "backoffFactor", r.backoffFactor)
```

Comment:
Retrier isn't exclusively used with kubectl wait, and I'm not certain if other retriable functions would log a timeout.

I added this because I wasn't clear, when reading a log, what the configured retrier timeout was.  I wanted that context while reading the log lines of the retrier's process. 

Also, kubectl wait timeout values are calculated based on remaining time.  While the first kubectl wait's timeout *should* be the exact value the retrier is configured with, I trust it less, given that its been recalculated.

Reply:
I'm ok with the log since its at level 5